pkg/reconciler/kubernetes/tektoninstallerset/client/obsolete_sets.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+func (i *InstallerSetClient) RemoveObsoleteSets(ctx context.Context) error {
+	var sets []string
+	if i.resourceKind == v1alpha1.KindTektonPipeline {
+		sets = []string{"pipeline", "PrePipeline", "PostPipeline"}
+	} else if i.resourceKind == v1alpha1.KindTektonTrigger {
+		sets = []string{"trigger"}
+	}
```

Comment:
Should we do it for all InstallerSets :thinking: i.e Dashboard, Rbac etc...

Reply:
those reconciler are not yet refactored.. once they are done we can do it for them